src/test/java/com/google/sps/MatchRankingTest.java

```diff
@@ -313,8 +318,68 @@ private static boolean compareDoubles(double d1, double d2) {
     return (Math.abs(d1 - d2) < 0.0001);
   }
 
+  @Test
+  public void testDifferentLengthBios() {
+    User currentUser = new User("");
+    addNewBio(user_a.getBio(), currentUser);
+    addNewBio(user_b.getBio(), currentUser);
+    addNewBio(user_d.getBio(), currentUser);
+
+    User userGood = new User("");
+    userGood.setId("09876");
+    User userBad = new User("");
+    userBad.setId("123456");
+
+    for(int i = 10; i < 5000; i += 50) {
+      try {
+        userGood.setBio(generateRandomBio(i, 1.0));
+        userBad.setBio(generateRandomBio(i, 0.0));
+
+        Collection<User> newMatches = new HashSet<User>();
+        newMatches.add(userGood);
+        newMatches.add(userBad);
+
+        List<User> result = MatchRanking.rankMatches(currentUser.getBioMap(), newMatches);
+        Assert.assertEquals(userGood, result.get(0));
+      } catch (Exception e) {
+        System.err.println("System could not handle bio of length " + i + ". See testDifferentLengthBios for more details.");
+        break;
+      }
+    }
+  }
+
+  //prints the percent keywords that pass with bio ranked at the top
```

Comment:
Nice work! 

For this test, the goal would be to validate that the rankMatches method does not get worse with future code changes. Could we also add some assertions based on the percentage?

Reply:
Done